alica_engine/include/engine/scheduler/Job.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "engine/AlicaClock.h"
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+namespace alica
+{
+namespace scheduler
+{
+struct Job
+{
+    int id;
+    std::function<void()> cb;
+    AlicaTime scheduledTime;
+    bool cancelled;
+    bool isRepeated;
+    AlicaTime repeatInterval;
+    std::vector<std::weak_ptr<Job>> prerequisites;
+
+    Job(int id, std::function<void()> cb, std::vector<std::weak_ptr<Job>> prerequisites)
+            : cb(cb)
+            , id(id)
+            , prerequisites(prerequisites)
+            , cancelled(false)
+            , isRepeated(false)
+    {}
+
+    bool isPrerequisite(const Job& other) const
+    {
+        for (std::weak_ptr<Job> jobWeakPtr : prerequisites) {
+            std::shared_ptr<Job> job = jobWeakPtr.lock();
+            if (job && *(job) == other) {
+                return true;
+            }
+        }
+        return false;
+    }
+
```

Comment:
move prerequisite checking to job.

int _prerequisiteIndex = 0;
isPrerequisiteFree()  
{
     // iterate over the prerequisites from _prerequisiteIndex, point _prereuisiteIndex to first unsatisfied prerequisite.
     if index < size of prerequisiteVector   return false
     else return true;
}